shardingsphere-scaling/shardingsphere-scaling-dialect/shardingsphere-scaling-opengauss/src/main/java/org/apache/shardingsphere/scaling/opengauss/component/checker/OpenGaussDataSourcePreparer.java

```diff
@@ -123,18 +133,51 @@ private String queryTableDefinition(final Connection sourceConnection, final int
                     case CREATE_TABLE:
                         sql = addIfNotExistsForCreateTableSQL(sql);
                         sql = replaceActualTableNameToLogicTableName(sql, each.getActualTableName(), each.getLogicTableName());
+                        sql = skipCreateTableExtendSet(sql);
                         return sql;
                     case ALTER_TABLE:
                         sql = replaceActualTableNameToLogicTableName(sql, each.getActualTableName(), each.getLogicTableName());
                         return sql;
-                    case UNKNOWN:
-                        return sql;
                     default:
-                        return sql;
+                        return "";
                 }
-            }).collect(Collectors.toList());
+            }).filter(sql -> !"".equals(sql)).collect(Collectors.toList());
             result.put(each.getLogicTableName(), logicTableSQLs);
         }
         return result;
     }
+    
+    @Override
+    protected String replaceActualTableNameToLogicTableName(final String createOrAlterTableSQL, final String actualTableName, final String logicTableName) {
+        StringBuilder sb = new StringBuilder(createOrAlterTableSQL);
+        while (true) {
+            int start = sb.indexOf(actualTableName);
+            if (start <= 0) {
+                return sb.toString();
+            }
+            int end = start + actualTableName.length();
+            sb.replace(start, end, logicTableName);
+        }
+    }
+    
+    private String skipCreateTableExtendSet(final String createSql) {
+        String lowerCreateSql = createSql.toLowerCase();
+        String[] search = {WITH_OF_TABLE_EXTEND, ")"};
+        List<Integer> searchPos = new ArrayList<>(2);
+        int startPos = 0;
+        for (String str: search) {
+            int curSearch = lowerCreateSql.indexOf(str, startPos);
+            if (curSearch <= 0) {
+                break;
+            }
+            searchPos.add(curSearch);
+            startPos = curSearch;
+        }
+        // if not search with ( and ), nothing to do.
```

Comment:
Could be removed.
Code of conduct: Split codes that need to add notes with it into small methods, which are explained with method names.